projects/objects/solids/protos/SolidRoundedBox.proto

```diff
@@ -17,258 +18,255 @@ PROTO SolidRoundedBox [
   field SFBool     enableBoundingObject  TRUE                                                                                                           # Defines whether the solid should have a bounding object.
 ]
 {
-  %{
-    -- parameter checking
-    local subdivision = fields.subdivision.value
-    if subdivision > 200 then
-      print "High value for 'subdivision'. This can slow down the simulation"
-    elseif subdivision <= 8 then
-      io.stderr:write("'subdivision' must be greater than 8\n")
-      subdivision = 8
-    end
+  %<
+    // parameter checking
+    let subdivision = fields.subdivision.value;
+    if (subdivision > 200)
+      console.log('High value for \'subdivision\'. This can slow down the simulation');
 
-    local borderRadius = fields.borderRadius.value
-    if borderRadius <= 0 then
-      io.stderr:write("'borderRadius' must be greater or equal to 0\n")
-      borderRadius = fields.borderRadius.defaultValue
-    end
+    if (subdivision < 8) {
+      console.error('\'subdivision\' must be greater or equal to 8. Using 8 instead.');
+      subdivision = 8;
+    }
 
-    local size = fields.size.value
-    if size.x <= 0 or size.y <= 0 or size.z <= 0 then
-      io.stderr:write("'size' must contain positive values\n")
-      size = fields.size.defaultValue
-    end
+    let borderRadius = fields.borderRadius.value;
+    if (borderRadius <= 0) {
+      borderRadius = fields.borderRadius.defaultValue;
+      console.error('\'borderRadius\' must be strictly positive. Value reset to ' + borderRadius + '.');
+    }
 
-    if 2 * borderRadius > size.x or
-       2 * borderRadius > size.y or
-       2 * borderRadius > size.z
-    then
-      io.stderr:write("'borderSize' must be smaller than the half of the 'size' elements\n")
-      borderSize = 0.5 * math.min(size.x, size.y, size.z)
-    end
+    let size = fields.size.value;
+    if (size.x <= 0 || size.y <= 0 || size.z <= 0) {
+      size = fields.size.defaultValue;
+      console.error('\'size\' must contain positive values. Value reset to (' + size.x + ', ' + size.y + ', ' + size.z + ').');
+    }
 
-    -- global stuff
-    borderDiameter = 2 * borderRadius
-    innerSize = { x = size.x - borderDiameter; y = size.y - borderDiameter; z = size.z - borderDiameter }
-  }%
+    if (2 * borderRadius > size.x || 2 * borderRadius > size.y || 2 * borderRadius > size.z) {
+      borderSize = 0.5 * Math.min(size.x, size.y, size.z);
```

Comment:
This creates an error as `borderSize` was not defined. This was already the case in the Lua version but did not raise an error as there is no control of that in Lua. I think it was only a typo in the first place and that you can replace `borderSize` by `borderRadius` (at l.45 and l.46).

Also, the error message (l.46) and the test effectuated (l.44) differs and by testing it it seems that the error should occurs when 2*`borderRadius` >= min(size) instead of border 2*`borderRadius` > min(size). Then, `borderRadius` should be reset to a value slightly smaller than min(size)/2. Otherwise, the SolidRoundedBox looks weird. Maybe this second issue can be addressed in another PR as it is a bug present already in the Lua version.